libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java

```diff
@@ -128,8 +130,54 @@ public void stop() {
         if (mFalsePositiveFilter.timeSinceInitialGPS() < TIME_UNTIL_GPS_GOES_COLD) {
             mFalsePositiveFilter.start();
         }
+
+        mNetworkLocationChangeDetector.stop();
+    }
+
+    static class NetworkLocationChangeDetector {
+        Location mLastLocation;
+        Context mContext;
+        private float DIST_THRESHOLD_M = 30.0f;
+
+        private LocationListener mNetworkLocationListener = new LocationListener() {
+            public void onLocationChanged(Location location) {
+                if (mLastLocation == null) {
+                    return;
+                }
+
+                if (location.distanceTo(mLastLocation) > DIST_THRESHOLD_M) {
+                    AppGlobals.guiLogInfo("MotionSensor.NetworkLocationChangeDetector triggered.");
+                    Intent sendIntent = new Intent(ACTION_USER_MOTION_DETECTED);
+                    LocalBroadcastManager.getInstance(mContext).sendBroadcastSync(sendIntent);
+                }
+            }
+
+            public void onStatusChanged(String provider, int status, Bundle extras) {}
+            public void onProviderEnabled(String provider) {}
+            public void onProviderDisabled(String provider) {}
+        };
+
+        public void start(Context c) {
+            mContext = c;
+            LocationManager lm = (LocationManager) mContext.getSystemService(Context.LOCATION_SERVICE);
+
+            if (mLastLocation == null) {
+                mLastLocation = lm.getLastKnownLocation(LocationManager.GPS_PROVIDER);
```

Comment:
Maybe change this to get whatever the highest accuracy location is by doing something like:
```java
matchingProviders = locationManager.getAllProviders();
for (String provider: matchingProviders) {
   location = locationManager.getLastKnownLocation(provider);
   
   // check that location is not null and grab the location with the best accuracy 
   // here
}
```